### PR TITLE
New/800/enclosure

### PR DESCRIPTION
**Pull request by jyothis-gireesan**
Endpoint support for Enclosure for V800

GET ALL
GET BY NAME
GET BY URI
CREATE
UPDATE
DELETE

### Check List
- [x] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in the examples (please include helpful comments).
  - [x] New endpoints supported are updated in the endpoints-support.md file.
- [x] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: endpoints-support.md [hunk 1]
```diff
@@ -20,6 +20,12 @@
 
 | Endpoints                                                                       | Verb     | V200 | V300 | V500 |V600 |V800
 | --------------------------------------------------------------------------------------- | -------- | :------------------: | :------------------: | :------------------: | :------------------: | :------------------: |
+|     **Enclosures**                                                                                                                          |
```
Comment (soodpr) [inline endpoints-support.md]: enclosure supports should be till api800

Comment (jyothis-gireesan) [reply endpoints-support.md]: changed.

File: ov/enclosure.go [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+		return enclosure, err
+	}
+	log.Debugf("GetEnclosure %s", data)
+	if err := json.Unmarshal([]byte(data), &enclosure); err != nil {
+		return enclosure, err
+	}
+	return enclosure, nil
+}
+
+func (c *OVClient) GetEnclosures(filter string, sort string) (EnclosureList, error) {
```
Comment (AnushaY1916) [inline ov/enclosure.go]: start and count query parameters are missing in GetEnclosures function

File: ov/enclosure.go [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+	log.Debugf("Response New Enclosure %s", data)
+	if err := json.Unmarshal([]byte(data), &t); err != nil {
+		t.TaskIsDone = true
+		log.Errorf("Error with task un-marshal: %s", err)
+		return err
+	}
+
+	//err = t.Wait()
+	if err != nil {
+		//	return err
```
Comment (AnushaY1916) [inline ov/enclosure.go]: Why is this line commented?

**Pull request merged**